exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java

```diff
@@ -116,6 +119,7 @@
   private boolean first = true;
   private int targetRecordCount;
   private final String fileName;
+  private Set<Path> currSpillDirs = new TreeSet<>();
```

Comment:
Agree with replacement to Guava but not sure with list. If we use list we will have the same number of paths in collection as number of temporary files.